lib/assets/javascripts/turbograft/turbohead.coffee

```diff
@@ -76,37 +101,37 @@ noDatasetMatchesIn = (attribute, collection) ->
 updateLinkTags = (activeDocument, newLinks) ->
   # style tag load events don't work in all browsers
   # as such we just hope they load ¯\_(ツ)_/¯
-  newLinks.forEach((linkNode) -> insertLinkTask(activeDocument, linkNode)())
-
-updateScriptTags = (activeDocument, newScripts, callback) ->
-  asyncSeries(
-    newScripts.map((scriptNode) -> insertScriptTask(activeDocument, scriptNode)),
-    callback
+  newLinks.forEach((linkNode) ->
+    newNode = linkNode.cloneNode()
+    activeDocument.head.appendChild(newNode)
+    triggerEvent("page:after-link-inserted", newNode)
   )
 
-asyncSeries = (tasks, callback) ->
-  return callback() if tasks.length == 0
-  task = tasks.shift()
-  task(-> asyncSeries(tasks, callback))
+updateScriptTags = (activeDocument, newScripts) ->
+  promise = Promise.resolve()
+  newScripts.forEach (scriptNode) ->
+    promise = promise.then(-> insertScript(activeDocument, scriptNode))
+  promise
+
+insertScript = (activeDocument, scriptNode) ->
+  url = scriptNode.src
+  if scriptPromises[url]
+    return scriptPromises[url]
 
-insertScriptTask = (activeDocument, scriptNode) ->
-  # We need to clone script tags in order to ensure that the browser executes them.
+  # Clone script tags to guarantee browser execution.
   newNode = activeDocument.createElement('SCRIPT')
   newNode.setAttribute(attr.name, attr.value) for attr in scriptNode.attributes
   newNode.appendChild(activeDocument.createTextNode(scriptNode.innerHTML))
-  insertAssetTask(activeDocument, newNode, 'script')
 
-insertLinkTask = (activeDocument, node) ->
-  insertAssetTask(activeDocument, node.cloneNode(), 'link')
-
-insertAssetTask = (activeDocument, newNode, name) ->
-  (done) ->
+  scriptPromises[url] = new Promise((resolve) ->
     onAssetEvent = (event) ->
-      triggerEvent("page:#{name}-error", event) if event.type == 'error'
+      triggerEvent("page:#script-error", event) if event.type == 'error'
```

Comment:
Hmm, that's an excellent point.  I *think* any scripts can be removed from `scriptPromises` at this point.  I'll let that soak overnight and make the change if I can't think of a problematic scenario.

To be clear, though, the error script will still be in `document.head`, so it'll never be reloaded.